fission/src/mirabuf/ScoringZoneSceneObject.ts

```diff
@@ -0,0 +1,276 @@
+import PreferencesSystem from "@/systems/preferences/PreferencesSystem"
+
+class ScoringZoneSceneObject extends SceneObject {
+    //Official FIRST hex
+    static redMaterial =  new THREE.MeshPhongMaterial({
+        color: 0xED1C24,
+        shininess: 0.0,
+        opacity: 0.7,
+        transparent: true,
+    })
+    static blueMaterial = new THREE.MeshPhongMaterial({
+        color: 0x0066B3,
+        shininess: 0.0,
+        opacity: 0.7,
+        transparent: true,
+    })  //0x0000ff
+    static transparentMaterial = new THREE.MeshPhongMaterial({
+        color: 0x0000,
+        shininess: 0.0,
+        opacity: 0.0,
+        transparent: true
+    })
+
+    private _parentAssembly: MirabufSceneObject
+    private _parentBodyId?: Jolt.BodyID
+    private _deltaTransformation?: THREE.Matrix4
+
+    private _toRender: boolean
+    private _prefs?: ScoringZonePreferences
+    private _joltBodyId?: Jolt.BodyID
+    private _mesh?: THREE.Mesh
+    private _collision?: (event: OnContactAddedEvent) => void
+    private _collisionRemoved?: (event: OnContactRemovedEvent) => void
+
+    private _gpContacted:  Jolt.BodyID[] = []
+
+    public get gpContacted() {
+        return this._gpContacted
+    }
+
+    public constructor(parentAssembly: MirabufSceneObject, index: number, render?: boolean) {
+        super()
+
+        console.debug("Trying to create scoring zone...")
+
+        this._parentAssembly = parentAssembly
+        this._prefs = this._parentAssembly.fieldPreferences?.scoringZones[index]
+        this._toRender = render ?? PreferencesSystem.getGlobalPreference<boolean>("RenderScoringZones")
+    }
+
+    public Setup(): void {
+        if (this._prefs) {
+            this._parentBodyId = this._parentAssembly.mechanism.nodeToBody.get(this._prefs.parentNode ?? this._parentAssembly.rootNodeId)
+
+            if (this._parentBodyId) {
+                // Create a default sensor
+                this._joltBodyId = World.PhysicsSystem.CreateSensor(
+                    new JOLT.BoxShapeSettings(
+                        new JOLT.Vec3(1,1,1)
+                    )
+                )
+                if (!this._joltBodyId) {
+                    console.log("Failed to create scoring zone. No Jolt Body")
+                    return
+                }
+                
+                // Position/rotate/scale sensor to settings
+                this._deltaTransformation = Array_ThreeMatrix4(this._prefs.deltaTransformation)
+                const fieldTransformation = JoltMat44_ThreeMatrix4(World.PhysicsSystem.GetBody(this._parentBodyId).GetWorldTransform())
+                const zoneTransformation = this._deltaTransformation.clone().premultiply(fieldTransformation)
+
+                const translation = new THREE.Vector3(0, 0, 0)
+                const rotation = new THREE.Quaternion(0, 0, 0, 1)
+                const scale = new THREE.Vector3(1, 1, 1)
+                zoneTransformation.decompose(translation, rotation, scale)
```

Comment:
I believe this same math is used two or three times, so maybe we could put it in a utils function. Not super important but would be nice if it changes down the line.